pkg/controller/volume/attachdetach/reconciler/reconciler.go

```diff
@@ -258,6 +268,8 @@ func (rc *reconciler) attachDesiredVolumes() {
 				klog.Infof(volumeToAttach.GenerateMsgDetailed("Volume attached--touching", ""))
 			}
 			rc.actualStateOfWorld.ResetDetachRequestTime(volumeToAttach.VolumeName, volumeToAttach.NodeName)
+			// remove any pre-existed cache entry in this case. The volume has been attached
+			rc.operationStartTimeCache.Delete(operationexecutor.CreateAttachDetachOperationUniqueKey(volumeToAttach.VolumeName, volumeToAttach.NodeName))
```

Comment:
should it be abort, or should we actually log the time?  What if attach operation starts, times out, and next time before we retry, it finishes?

Reply:
It's abort here to avoid leaking in cases when the volume has been attached out of bound while the controller is trying.
for the case "What if attach operation starts, times out, and next time before we retry, it finishes?", it will be a no-op. The operation-generator change reports the metrics before asw is updated.

Reply:
Are we sure that ASW cannot be updated through some other operation like VerifyVolumesAreAttached?

Would it be safer to just record metric anyway?  If it was already recorded by operation generator, then this should be a noop right?